### PR TITLE
Remove sygus print callback

### PR DESCRIPTION
**Pull request by ajreynol**
This removes sygus print callbacks, since they are no longer necessary to support the sygus v1 parser.

This involved generalizing the scope of the datatype utility sygusToBuiltin.  Now, printing "as sygus" simply is accomplished by doing sygusToBuiltin conversion and then calling the printer on the builtin term.

This is required for further work towards eliminating the Expr layer.

FYI @4tXJ7f

### CODE REVIEW

File: src/theory/datatypes/sygus_datatype_utils.cpp [hunk 1]
```diff
@@ -343,6 +351,9 @@ Node sygusToBuiltin(Node n, bool isExternal)
     it = visited.find(cur);
     if (it == visited.end())
     {
+      // notice this condition succeeds in roughly 99% of the executions of this
```
Comment (HanielB) [inline src/theory/datatypes/sygus_datatype_utils.cpp:354]: Could you give more details on this number? For typical benchmarks, regressions...? Circa 20XX?

File: src/theory/datatypes/sygus_datatype_utils.cpp [hunk 2]
```diff
@@ -359,6 +370,27 @@ Node sygusToBuiltin(Node n, bool isExternal)
...
+          // use the previously constructed variable for it
+          visited[cur] = cur.getAttribute(SygusToBuiltinVarAttribute());
+        }
+        else
+        {
+          std::stringstream ss;
+          ss << cur;
+          const DType& dt = cur.getType().getDType();
+          // make a fresh variable
+          NodeManager * nm = NodeManager::currentNM();
```
Comment (HanielB) [inline src/theory/datatypes/sygus_datatype_utils.cpp:387]:
```suggestion
          NodeManager* nm = NodeManager::currentNM();
```
Moreover move this to outside the loop.

File: src/theory/datatypes/sygus_datatype_utils.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/datatypes/sygus_datatype_utils.cpp:354]: Coverage tests :)  They are helpful.

Comment (HanielB) [reply src/theory/datatypes/sygus_datatype_utils.cpp]: Ah, nice. :)

File: src/theory/datatypes/sygus_datatype_utils.cpp (hunk 2 above)
Comment (ajreynol) [reply src/theory/datatypes/sygus_datatype_utils.cpp:387]: Its intentionally here for performance reasons (this is partly the point of the comment mentioned above).

**Pull request merged**